### PR TITLE
fix(arrow): union buffer count & handle schema errors

### PR DESCRIPTION
**Pull request by Mause**
Fixes two small issues I found while adding nested type support to duckdb-rs

### CODE REVIEW

**Comment by Mause** [comment]
> You should probably look at `feature` to see if there are conflicts there, we have made some changes to the Arrow code on feature

Looks like there will be a single line conflict yeah

File: src/common/arrow/appender/union_data.cpp [hunk 1]
```diff
@@ -55,8 +55,8 @@ void ArrowUnionData::Append(ArrowAppendData &append_data, Vector &input, idx_t f
 }
 
 void ArrowUnionData::Finalize(ArrowAppendData &append_data, const LogicalType &type, ArrowArray *result) {
-	result->n_buffers = 2;
-	result->buffers[1] = append_data.main_buffer.data();
+	result->n_buffers = 1;
+	result->buffers[0] = append_data.main_buffer.data();
```
Comment (Tishj) [inline src/common/arrow/appender/union_data.cpp:59]: Isn't the first buffer always occupied by the validity data?
I'm not sure I understand this change

Comment (Mause) [reply src/common/arrow/appender/union_data.cpp:59]: > All array types, with the exception of union types (more on these later), utilize a dedicated memory buffer, known as the validity (or “null”) bitmap, to encode the nullness or non-nullness of each value slot. The validity bitmap must be large enough to have at least 1 bit for each array slot.

https://arrow.apache.org/docs/format/Columnar.html#validity-bitmaps

Comment (Mause) [reply src/common/arrow/appender/union_data.cpp:59]: This tripped me up as well when I wrote the original implementation, but I guess most readers are pretty forgiving

Comment (Tishj) [reply src/common/arrow/appender/union_data.cpp:59]: Ah actually, yea this makes sense: https://arrow.apache.org/docs/format/Columnar.html#buffer-listing-for-each-layout

**Review by Tishj — APPROVED** [review]
Looks good, thanks for fixing this!

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**